src/Types/EnumType.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Types;
+
+use BackedEnum;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Platforms\PostgreSQLPlatform;
+use Doctrine\DBAL\Platforms\SQLitePlatform;
+use Doctrine\DBAL\Platforms\SQLServerPlatform;
+use Doctrine\DBAL\Types\Exception\InvalidFormat;
+use Doctrine\DBAL\Types\Exception\InvalidType;
+use Doctrine\DBAL\Types\Exception\ValueNotConvertible;
+use ReflectionClass;
+use Stringable;
+use Throwable;
+use UnitEnum;
+
+use function array_map;
+use function class_exists;
+use function enum_exists;
+use function implode;
+use function in_array;
+use function is_object;
+use function is_string;
+use function sprintf;
+
+final class EnumType extends Type
+{
+    public string $name = 'enum';
+
+    public ?string $enumClassname = null;
+
+    /** @var array<int, string> */
+    public array $members = [];
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getMappedDatabaseTypes(AbstractPlatform $platform): array
+    {
+        return [$this->name];
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getSQLDeclaration(array $column, AbstractPlatform $platform): string
+    {
+        $values = implode(
+            ', ',
+            array_map(
+                static fn (string $value) => sprintf('\'%s\'', $value),
+                $column['members'] ?: $column['type']->members,
+            ),
+        );
+
+        return match (true) {
+            $platform instanceof SQLitePlatform => sprintf('TEXT CHECK(%s IN (%s))', $column['name'], $values),
+            $platform instanceof PostgreSQLPlatform, $platform instanceof SQLServerPlatform => sprintf('VARCHAR(255) CHECK(%s IN (%s))', $column['name'], $values),
+            default => sprintf('ENUM(%s)', $values),
+        };
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function convertToDatabaseValue($value, AbstractPlatform $platform): ?string
```

Comment:
```suggestion
    public function convertToDatabaseValue(mixed $value, AbstractPlatform $platform): ?string
```

Why did your remove the parameter type?